plugin-adscc/src/main/java/org/apache/ranger/services/adscc/RangerServiceAdscc.java

```diff
@@ -0,0 +1,128 @@
+package org.apache.ranger.services.adscc;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.log4j.Logger;
+import org.apache.ranger.plugin.model.RangerService;
+import org.apache.ranger.plugin.model.RangerServiceDef;
+import org.apache.ranger.plugin.service.RangerBaseService;
+import org.apache.ranger.plugin.service.ResourceLookupContext;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RangerServiceAdscc extends RangerBaseService {
+    private static final Logger logger = Logger.getLogger(RangerServiceAdscc.class);
+    private static final String HOST_CONFIG = "hostname";
+    private static final String USERNAME_CONFIG = "username";
+    private static final String PASSWORD_CONFIG = "password";
+    private static final String SSL_TRUSTSTORE_CONFIG = "ssl_truststore";
+    private static final String SSL_TRUSTSTORE_TYPE_CONFIG = "ssl_truststore_type";
+    private static final String SSL_TRUSTSTORE_PASSWORD_CONFIG = "ssl_truststore_password";
+
+
+    private AdsccRestService adsccRestService;
+    private RangerAdsccLookupService rangerAdsccLookupService;
+
+    public RangerServiceAdscc() {
+        super();
+    }
+
+    @Override
+    public void init(final RangerServiceDef serviceDef, final RangerService service) {
+        super.init(serviceDef, service);
+        adsccRestService = new AdsccRestService(getHttpClient());
+        rangerAdsccLookupService = new RangerAdsccLookupService(adsccRestService);
+    }
+
+    @Override
+    public Map<String, Object> validateConfig() throws Exception {
+        Map<String, Object> result = new HashMap<>();
+        try {
+            String hostname = configs.get(HOST_CONFIG);
+            if (hostname.lastIndexOf("/") == hostname.length() - 1) {
+                result.put(HOST_CONFIG, "Incorrect format, remove last character /");
+            } else {
+                int statusCode = AdsccEntityEnum.CLUSTER.getUrl(new HashMap<>()).map(url -> adsccRestService.getStatusCode(configs.get(HOST_CONFIG) + url,
```

Comment:
format it please, so that it won't be so long

Reply:
Fixed.